app/uk/gov/hmrc/apisubscriptionfields/logging/ApiLogger.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.logging
+
+import javax.inject.Singleton
+
+import com.google.inject.Inject
+import uk.gov.hmrc.play.config.inject.ServicesConfig
+
+@Singleton
+class ApiLogger @Inject()(serviceConfig: ServicesConfig) {
```

Comment:
Is this really needed?
It seems a useless wrapper in my opinion.
Why don't you use directly `play.api.Logger` in the code?

Reply:
@gokyo  we use a class like the `ApiLogger` wrapper in Customs for the following reasons:
- separation of service logging concerns from play logging (default is to use play logger) 
    - eg `play.api.Logger.info("usefull stuff...")` uses play logger so mixed in with low level Play internals stuff
    - see config change https://github.com/googley42/api-subscription-fields/blob/API-2532_pr3_squashed/conf/application-json-logger.xml#L10
    - here we assign a new logger `api-subscription-fields` for application logging
    - this enables us to filter on `api-subscription-fields` logger in KABANA
    - we can change service logging levels (eg DEBUG) in isolation to Play internal logging
- `ApiLogger` wrapper avoids repition of code like `var logger = play.api.Logger("api-subscription-fields")`
- opens up possibilities to test really important logging using mocks eg in functions that return `Unit`
- lastly having a wrapper like `ApiLogger` makes sure that logging calls are excluded from code coverage

Reply:
I reviewed this with @AndySpaven and @paulblakehmrc 
We do not see advantages on doing this.

Our suggestion is:
- Use directly the `Play` logger in all the code.
- You do not need to define a `val logger = play.api.Logger` in all classes, you can just use directly the companion object and call the logging methods. See here: https://github.com/hmrc/api-gateway/blob/master/app/uk/gov/hmrc/apigateway/controller/ProxyController.scala
- You can specify the logging level for your application and its dependencies (Play, Netty, Mongo, etc) in the `application-json-logger.xml` file. See this example: https://github.com/hmrc/api-gateway/blob/master/conf/application-json-logger.xml
- The logging that is relevant to you can be searched in Kibana with this type of query: `app:"api-subscription-fields" AND logger="application" AND level:INFO`
- Do not worry to call the Play logger with the name `api-subscription-fields`, let's leave the name `application`